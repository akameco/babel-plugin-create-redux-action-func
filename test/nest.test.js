// @flow
import transformCode from './transform'

test('nest union type', () => {
  const input = `
    type HOGE_TYPE = 'hoge/HOGE_ACTION'
    type FOO_TYPE = 'hoge/FOO_ACTION'
    type Action = { +type: HOGE_TYPE | FOO_TYPE }
  `
  expect(transformCode(input)).toMatchSnapshot()
})

test('nest union type with args', () => {
  const input = `
    type HOGE_TYPE = 'hoge/HOGE_ACTION'
    type FOO_TYPE = 'hoge/FOO_ACTION'
    type FOO_FUGA_TYPE = 'hoge/FOO_ACTION'
    type Action =
      | { +type: HOGE_TYPE | FOO_TYPE }
      | { +type: FOO_FUGA_TYPE, id: number }
  `
  expect(transformCode(input)).toMatchSnapshot()
})
