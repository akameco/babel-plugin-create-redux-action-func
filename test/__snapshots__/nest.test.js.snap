// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`another type import 1`] = `
"// @flow
import type { Action } from './actionTypes';
import { FOO, FOO_ID } from './constants';

import type { ID, Text } from './types';

export function foo(): Action {
  return {
    type: FOO
  };
}
export function fooId(id: ID, text: Text): Action {
  return {
    type: FOO_ID,
    id,
    text
  };
}"
`;

exports[`nest union type 1`] = `
"// @flow
import type { Action } from './actionTypes';
import { HOGE, FOO } from './constants';

export function hoge(): Action {
  return {
    type: HOGE
  };
}
export function foo(): Action {
  return {
    type: FOO
  };
}"
`;

exports[`nest union type with args 1`] = `
"// @flow
import type { Action } from './actionTypes';
import { HOGE, FOO, FOO_FUGA } from './constants';

export function hoge(): Action {
  return {
    type: HOGE
  };
}
export function foo(): Action {
  return {
    type: FOO
  };
}
export function fooFuga(id: number): Action {
  return {
    type: FOO_FUGA,
    id
  };
}"
`;
